include/actuator/ActuatorNGP.h

```diff
@@ -0,0 +1,87 @@
+// Copyright 2017 National Technology & Engineering Solutions of Sandia, LLC
+// (NTESS), National Renewable Energy Laboratory, University of Texas Austin,
+// Northwest Research Associates. Under the terms of Contract DE-NA0003525
+// with NTESS, the U.S. Government retains certain rights in this software.
+//
+// This software is released under the BSD 3-clause license. See LICENSE file
+// for more details.
+//
+
+#ifndef ACTUATOR_NGP_H_
+#define ACTUATOR_NGP_H_
+
+#include <actuator/ActuatorTypes.h>
+#include <stk_mesh/base/BulkData.hpp>
+
+namespace sierra{
+namespace nalu{
+
+
+/*! \struct AcutatorFunctor
+ *  \brief Generalized functor for performing actuator work
+ *
+ * The goal of this functor is to generalize the interface
+ * for doing work on host/device
+ *
+ * \tparam BulkData The type of bulk data that this functor will execute on
+ *  (i.e. Actuator line w/h FAST vs a generalized instance)
+ *
+ * \tparam TAG A TAG to distinguish instances of operations
+ *  (i.e. preIter vs findPoints)
+ */
+
+template<typename BulkData, typename TAG, typename ExecutionSpace>
```

Comment:
So far in the usage `TAG` seems unnecessary. No `operator()(...)` seems to really need it since every operation is its own unique functor.

Reply:
@rcknaus had the same comment. The `TAG` is there so the objects don't need to be constructed, but rather the constructor can be passed in the `kokkos::parallel_for`.  I can create them inside the `execute` function of the `Actuator` class though and get similar behavior without the `TAG`.